File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer> map = new HashMap<>();
        map.put(0,1);
        int ans = 0;
        int prefSum=0;
        for(int i : nums) {
            prefSum += i;
            ans += map.getOrDefault(prefSum-k,0);
            map.put(prefSum,map.getOrDefault(prefSum,0)+1);
        }
        return ans;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        map[0]++;
        int ans = 0;
        int prefSum=0;
        for(int i : nums) {
            prefSum+= i;
            if(map.find(prefSum-k)!=map.end()) {
                ans+=map[prefSum-k];
            }
            map[prefSum]++;
        }
        return ans;
    }
};